src/zope/sqlalchemy/tests.py

```diff
@@ -66,74 +66,68 @@ class Skill(SimpleModel):
 
 # See https://code.google.com/p/pysqlite-static-env/
 HAS_PATCHED_PYSQLITE = False
-if engine.url.drivername == 'sqlite':
+if engine.url.drivername == "sqlite":
     try:
         from pysqlite2.dbapi2 import Connection
     except ImportError:
         pass
     else:
-        if hasattr(Connection, 'operation_needs_transaction_callback'):
+        if hasattr(Connection, "operation_needs_transaction_callback"):
             HAS_PATCHED_PYSQLITE = True
 
 if HAS_PATCHED_PYSQLITE:
     from sqlalchemy import event
     from zope.sqlalchemy.datamanager import NO_SAVEPOINT_SUPPORT
-    NO_SAVEPOINT_SUPPORT.remove('sqlite')
 
-    @event.listens_for(engine, 'connect')
+    NO_SAVEPOINT_SUPPORT.remove("sqlite")
+
+    @event.listens_for(engine, "connect")
     def connect(dbapi_connection, connection_record):
         dbapi_connection.operation_needs_transaction_callback = lambda x: True
 
 
-Session = orm.scoped_session(orm.sessionmaker(
-    bind=engine,
-    extension=tx.ZopeTransactionExtension(),
-    twophase=TEST_TWOPHASE,
-))
+Session = orm.scoped_session(orm.sessionmaker(bind=engine, twophase=TEST_TWOPHASE))
+tx.register(Session)
 
-UnboundSession = orm.scoped_session(orm.sessionmaker(
-    extension=tx.ZopeTransactionExtension(),
-    twophase=TEST_TWOPHASE,
-))
+UnboundSession = orm.scoped_session(orm.sessionmaker(twophase=TEST_TWOPHASE))
+tx.register(UnboundSession)
 
-EventSession = orm.scoped_session(orm.sessionmaker(
-    bind=engine,
-    twophase=TEST_TWOPHASE,
-))
+EventSession = orm.scoped_session(orm.sessionmaker(bind=engine, twophase=TEST_TWOPHASE))
+tx.register(EventSession)
 
-KeepSession = orm.scoped_session(orm.sessionmaker(
-    bind=engine,
-    extension=tx.ZopeTransactionExtension(keep_session=True),
-    twophase=TEST_TWOPHASE,
-))
+KeepSession = orm.scoped_session(orm.sessionmaker(bind=engine, twophase=TEST_TWOPHASE))
+tx.register(KeepSession, keep_session=True)
 
-tx.register(EventSession)
```

Comment:
I'm not going to look at the diff of this file, because it's difficult to notice actual changes amidst all the formatting changes.

Reply:
Ah damn... I personally don't use black but I was under the impression it was used in here ... I just slapped it on after my and didn't look too closely ... sigh. 

So, if anyone wants me to change this back then I shall take some time and "unblack" it ... (although that's also killing time I don't have, but I've been naughty, so ...)

Reply:
No, it's fine, the bugfix is too important to get bogged on formatting details, let's just merge the fix, I'm sure the tests are _fine_.

Reply:
I promise I ran them and got a green bar ;)